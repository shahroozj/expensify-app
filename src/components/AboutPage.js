import React from 'react';

const AboutPage = () =>(
    <div>
      <h1>About Me</h1>
      <p>Hi, My Name is Shahrooz</p>
    </div>
)

export default AboutPage;