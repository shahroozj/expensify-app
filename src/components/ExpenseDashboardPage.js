import React from 'react';
import ExpenseList from './ExpenseList';

const ExpenseDashboardPage = () =>(
    <div>
      This is dashboard of the application

      <ExpenseList />
    </div>
)

export default ExpenseDashboardPage;