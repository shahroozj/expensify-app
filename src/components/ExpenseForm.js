import React from 'react';

export default class ExpenseFrom extends React.Component{
    state = {
        description : '',
        note : '',
        amount : ''
    }

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({description}));
    }

    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({note}));
    }

    onAmountChange = (e) => {
        const amount = e.target.value;
        if(amount.match(/^\d*(\.\d{0,2})?$/)){
            this.setState(() => ({amount}));
        }
    }

    render(){
        return(
            <div>
                <form>
                    <input 
                    type = "text"
                    placeholder = "Description"
                    autoFocus
                    value = {this.state.description}
                    onChange = {this.onDescriptionChange}
                    >
                    </input>
                    <input
                    type="text"
                    placeholder="Amount"
                    value = {this.state.amount}
                    onChange = {this.onAmountChange}
                    >
                    </input>
                    <textarea
                    placeholder="Add note for this expense (optional)"
                    value = {this.state.note}
                    onChange = {this.onNoteChange}
                    ></textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }

}