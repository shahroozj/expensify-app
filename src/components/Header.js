import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';

export const Header = ({startLogout}) => (
    <header>
        <h1>
            Expensify
        </h1>
        <ul>
            <li><NavLink to="/about" activeClassName="is-active">About</NavLink></li>
            <li><NavLink to="/dashboard" activeClassName="is-active" exact={true}>Dashboard</NavLink></li>
            <li><NavLink to="/create" activeClassName="is-active">Create Expense</NavLink></li>
            <li><NavLink to="/help" activeClassName="is-active">Help</NavLink></li>
            <button onClick={startLogout}>Logout</button>
        </ul>
    </header>
);

const mapDispatchToProps = (dispatch) =>({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);