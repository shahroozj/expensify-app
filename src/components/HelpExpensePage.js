import React from 'react';

const HelpExpensePage = () =>(
    <div>
      This is Help page
    </div>
);

export default HelpExpensePage;
