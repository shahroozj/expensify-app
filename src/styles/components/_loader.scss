.loader{
    align-items: center;
    justify-content: center;
    display: flex;
    height: 100vh;
    widows: 100vw;
}

.loader__image{
    width: 6rem;
    height: 6rem;
}