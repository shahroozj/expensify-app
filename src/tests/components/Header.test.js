import React from 'react';
import {shallow} from 'enzyme';
//import toJSON from 'enzyme-to-json'; we removed this import because we set it up in config.
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import {Header} from '../../components/Header';

test('should render header correctly', () => {
    const wrapper = shallow(<Header startLogout={()=>{}}/>);
    //expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();
    //expect(wrapper.find('h1').text()).toBe('Expensify');
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout} />);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
});