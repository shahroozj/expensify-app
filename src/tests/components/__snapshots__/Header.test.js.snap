// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/about"
      >
        About
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        exact={true}
        to="/"
      >
        Dashboard
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/create"
      >
        Create Expense
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/help"
      >
        Help
      </NavLink>
    </li>
  </ul>
</header>
`;
